src/fides/api/service/connectors/scylla_query_config.py

```diff
@@ -0,0 +1,92 @@
+import re
+from typing import Any, Dict, List, Optional, Tuple
+
+from fides.api.graph.config import Field
+from fides.api.models.policy import Policy
+from fides.api.service.connectors.query_config import SQLLikeQueryConfig
+
+ScyllaDBStatement = Tuple[str, Dict[str, Any]]
+"""
+A ScyllaDB statement is a tuple whose first component is the query string
+and whose second component is a dict of parameters to pass in to the query.
+
+E.g
+('SELECT name FROM users WHERE user.id = %(user_id)s', { "user_id": 123 })
+"""
+
+
+class ScyllaDBQueryConfig(SQLLikeQueryConfig[ScyllaDBStatement]):
+    def query_to_str(
+        self, t: ScyllaDBStatement, input_data: Dict[str, List[Any]]
+    ) -> str:
+        """string representation of a query for logging/dry-run"""
+
+        def transform_param(p: Any) -> str:
+            if isinstance(p, str):
+                return f"'{p}'"
+            return str(p)
+
+        query_str = str(t[0])
+        query_params = t[1]
+
+        for k, v in query_params.items():
+            query_str = re.sub(rf"%\({k}\)s", f"{transform_param(v)}", query_str)
+
+        return query_str
+
+    def dry_run_query(self) -> Optional[str]:
+        """Returns a text representation of the query."""
+        query_data = self.display_query_data()
+        query_statement = self.generate_query(query_data, None)
+        if not query_statement:
+            return None
+
+        return self.query_to_str(query_statement, query_data)
+
+    def format_clause_for_query(
+        self, string_path: str, operator: str, operand: str
+    ) -> str:
+        """Returns clauses in a format they can be added into ScyllaDB CQL queries."""
+        if operator == "IN":
+            return f"{string_path} IN ({operand})"
+        return f"{string_path} {operator} %({operand})s"
+
+    def get_formatted_query_string(
+        self,
+        field_list: str,
+        clauses: List[str],
+    ) -> str:
+        """Returns an SQL query string."""
+        return f"SELECT {field_list} FROM {self.node.collection.name} WHERE {' OR '.join(clauses)} ALLOW FILTERING;"
```

Comment:
CQL does not encourage you to filter on any column in a `WHERE` clause -- rather it only wants you to filter on clustering columns (i.e columns that belong to a primary key). In order to bypass this restriction, we need to add `ALLOW FILTERING` to the query. This could have a potentially large performance impact: if the columns on the `WHERE` clause are not clustering columns and don't have any secondary indexes created, the query may be slow, particularly if the table has a lot of rows.

The following is quoted from [the ScyllaDB docs](https://opensource.docs.scylladb.com/stable/cql/dml/select.html#allowing-filtering): 
> Please note that a query using ALLOW FILTERING may thus have unpredictable performance (for the definition above), i.e. even a query that selects a handful of records may exhibit performance that depends on the total amount of data stored in the cluster.